### PR TITLE
ci: check GHA results in TeamCity

### PR DESCRIPTION
**Pull request by rail**
Bazel Essential CI is a no-op on master and release-24.1 since the GitHub Actions CI migration completed.

This PR adds a new TeamCity build configuration which polls the required GitHub action results and reflects the overall status in TeamCity.

Fixes: DEVINF-1124
Release note: None

### CODE REVIEW

**Comment by rail** [comment]
I added a new TC configuration and added it to the MPB deps.
Examples https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_Tests_CheckGhaResults?branch=123421&buildTypeTab=overview&mode=builds

File: pkg/cmd/github-action-poller/main.go [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+	}
+}
+
+func run(ctx context.Context, owner, repo, sha string, requiredChecks []string) (bool, error) {
+	statuses := make(map[string][]*github.CheckRun, len(requiredChecks))
+	for _, c := range requiredChecks {
+		statuses[c] = nil
+	}
+
+	client := github.NewClient(nil)
```
Comment (rickystewart) [inline pkg/cmd/github-action-poller/main.go:97]: This is unauthenticated, right? We'll hit rate limits even more quickly than usual. Maybe it doesn't make a difference if we do few enough requests.

**Review by rickystewart — APPROVED** [review]

File: pkg/cmd/github-action-poller/main.go (hunk 1 above)
Comment (rail) [reply pkg/cmd/github-action-poller/main.go:97]: Correct. I can easily add auth if we want. Actually, it was there in the initial version. :)

**Comment by rail** [comment]
bors r=rickystewart

**Comment by rail** [comment]
bors r=rickystewart

**Pull request merged**